### PR TITLE
ENH: Resample BOLD to fsLR directly, dropping fsaverage intermediate

### LINKED ISSUE #2990: good voxels masking error
### What happened?

I was able to run the goodvoxels flag though on fmriprep 23.0.1 and compared the same data to fmriprep 23.0.1 w/o goodvoxels and abcd-dcan 0.1.3

(the abcd, 23.0.1 (no good voxels arg) and the 23.0.1 goodvoxels result are attached. )
There seems to be a masking error on the surface with the goodvoxels.

### What command did you use?

```shell
--participant-label xxxx --output-spaces T1w MNI152NLin2009cAsym:res-2 --cifti-output --output-layout legacy --project-goodvoxels
```

### What version of fMRIPrep are you running?

23.0.1

### How are you running fMRIPrep?

Docker

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
[environment]
cpu_count = 24
exec_env = "docker"
free_mem = 82.8
overcommit_policy = "heuristic"
overcommit_limit = "50%"
nipype_version = "1.8.5"
templateflow_version = "23.0.0"
version = "23.0.1"

[execution]
bids_dir = "/input"
bids_database_dir = "/tmp/work/20230410-172026_c909688d-781d-414e-a9ca-062d24a43f68/bids_db"
bids_description_hash = "f7837453d53f5e81fac160140151ed912fdd1da09e92d853373c6d152bfdae5e"
boilerplate_only = false
sloppy = false
debug = []
fmriprep_dir = "/output/fmriprep"
fs_license_file = "/opt/freesurfer/license.txt"
fs_subjects_dir = "/output/freesurfer"
layout = "BIDS Layout: .../input | Subjects: 4 | Sessions: 4 | Runs: 15"
log_dir = "/output/fmriprep/logs"
log_level = 25
low_mem = false
md_only_boilerplate = false
notrack = false
track_carbon = false
country_code = "CAN"
output_dir = "/output"
me_output_echos = false
output_layout = "legacy"
output_spaces = "T1w MNI152NLin2009cAsym:res-2"
reports_only = false
run_uuid = "20230410-172026_c909688d-781d-414e-a9ca-062d24a43f68"
participant_label = [ "patensasc0089",]
templateflow_home = "/home/fmriprep/.cache/templateflow"
work_dir = "/tmp/work"
write_graph = false

[workflow]
anat_only = false
aroma_err_on_warn = false
aroma_melodic_dim = -200
bold2t1w_dof = 6
bold2t1w_init = "register"
cifti_output = "91k"
fmap_bspline = false
force_syn = false
hires = true
ignore = []
longitudinal = false
medial_surface_nan = false
project_goodvoxels = true
regressors_all_comps = false
regressors_dvars_th = 1.5
regressors_fd_th = 0.5
run_reconall = true
skull_strip_fixed_seed = false
skull_strip_template = "OASIS30ANTs"
skull_strip_t1w = "force"
slice_time_ref = 0.5
spaces = "T1w MNI152NLin2009cAsym:res-2 fsaverage:den-164k MNI152NLin6Asym:res-2"
use_aroma = false
use_syn_sdc = false

[nipype]
crashfile_format = "txt"
get_linked_libs = false
nprocs = 24
omp_nthreads = 8
plugin = "MultiProc"
resource_monitor = false
stop_on_first_crash = false

[seeds]
master = 27307
ants = 55820
numpy = 55599

[nipype.plugin_args]
maxtasksperchild = 1
raise_insufficient = false
```

### Additional information / screenshots

abcd
23 0 1nogoodvoxels
goodvoxels

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

1) Remove goodvoxels and CIFTI-specific code from `mri_vol2surf` resampling workflow. fsnative and fsaverage will remain the same for now.
2) Removes `fsaverage` as an implicit resampling target for `--cifti-output`. `fsaverage` and `fsLR` are now entirely independent.
3) Create a subject-specific cortical ROI mask in GIFTI space. Not currently output, as it's only needed internally; it's not a great mask but is intended to be more liberal than the fsLR mask applied later.
4) Adds a number of `wb_command` commands, including an OpenMP mixin that allows `OMP_NUM_THREADS` to be set in the environment. I think most commands can take advantage, but I only tagged slow ones (take >10s) that I could verify actually used them.
5) Depends on https://github.com/nipreps/smriprep/pull/339 and https://github.com/nipreps/niworkflows/pull/806.

TODO:

* [x] Restore literate workflow code.
* [x] Restore saving goodvoxels mask, if desired.

## Documentation that should be reviewed

TODO

---

Closes #2990.

### CODE REVIEW

**Comment by effigies** [comment]
@mgxd I just realized this hasn't been reviewed. Any chance you could have a look? I'll work on knocking off those two checklist points above.

File: fmriprep/workflows/bold/resampling.py [hunk 1]
```diff
@@ -242,65 +216,25 @@ def select_target(subject_id, space):
...
-            (goodvoxels_bold_mask_wf, rename_src, [("outputnode.masked_bold", "in_file")]),
-            (goodvoxels_bold_mask_wf, outputnode, [
-                ("outputnode.goodvoxels_ribbon", "goodvoxels_ribbon"),
-            ]),
-            (sampler, metric_dilate, [("out_file", "in_file")]),
-        ])
-        # fmt: on
-    else:
-        workflow.connect([(inputnode, rename_src, [("source_file", "in_file")])])
+    workflow.connect([(inputnode, rename_src, [("source_file", "in_file")])])
```
Comment (mgxd) [inline fmriprep/workflows/bold/resampling.py:228]: personal styling: seems wasteful to construct these objects for one connection

```suggestion
    workflow.connect(inputnode, "source_file", rename_src, "in_file")
```

File: fmriprep/workflows/bold/resampling.py [hunk 2]
```diff
@@ -558,29 +492,210 @@ def _calc_lower_thr(in_stats):
...
-            (apply_goodvoxels_ribbon_mask, outputnode, [("out_file", "masked_bold")]),
+            (goodvoxels_mask, outputnode, [("out_file", "goodvoxels_mask")]),
             (goodvoxels_ribbon_mask, outputnode, [("out_file", "goodvoxels_ribbon")]),
         ]
     )
 
     return workflow
 
 
+def init_bold_fsLR_resampling_wf(
```
Comment (mgxd) [inline fmriprep/workflows/bold/resampling.py:508]: this is a really nice digestion of the wb commands used

**Review by mgxd — COMMENTED** [review]
Looks good, skimmed through and nothing jumps out.

One note - the more that this can be upstreamed, the better - since we'll be using them in nibabies as well.

**Comment by effigies** [comment]
> One note - the more that this can be upstreamed, the better - since we'll be using them in nibabies as well.

Do you mean push interfaces into niworkflows?

**Comment by mgxd** [comment]
Yes, but it was not to say that should hold this up - we can port it over for `1.8.1`, cut a nwf release, and then remove it from here.

File: fmriprep/workflows/bold/resampling.py [hunk 3]
```diff
@@ -558,29 +487,277 @@ def _calc_lower_thr(in_stats):
...
+
+    Workflow Graph
+        .. workflow::
+            :graph2use: colored
+            :simple_form: yes
+
+            from fmriprep.workflows.bold.resampling import init_bold_fsLR_resampling_wf
+            wf = init_bold_fsLR_resampling_wf(
+                estimate_goodvoxels=True,
+                grayord_density='92k',
```
Comment (feilong) [inline fmriprep/workflows/bold/resampling.py:525]: Perhaps `grayord_density='91k'`  instead of 92k based on the Parameters below?

**Pull request merged**